ray-operator/apis/ray/v1alpha1/rayservice_types.go

```diff
@@ -7,16 +7,76 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type ServiceStatus string
+
+const (
+	FailToGetRayService          ServiceStatus = "FailToGetRayService"
+	FailToGetOrCreateRayCluster  ServiceStatus = "FailToGetOrCreateRayCluster"
+	WaitForDashboard             ServiceStatus = "WaitForDashboard"
+	FailServeDeploy              ServiceStatus = "FailServeDeploy"
+	FailGetServeDeploymentStatus ServiceStatus = "FailGetServeDeploymentStatus"
+	Running                      ServiceStatus = "Running"
+	Restarting                   ServiceStatus = "Restarting"
+	FailDeleteRayCluster         ServiceStatus = "FailDeleteRayCluster"
+)
+
 // RayServiceSpec defines the desired state of RayService
 type RayServiceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+	ServeConfigSpecs []ServeConfigSpec `json:"serveConfigs,omitempty"`
```

Comment:
Curious the slide here. Does one service allow multiple service configs?

Reply:
1 service can have multiple serve deployment.
Each `ServeConfigSpec` is for a serve deployment.